test/core/client_channel/http_proxy_mapper_test.cc

```diff
@@ -70,6 +70,121 @@ TEST(NoProxyTest, EmptyEntries) {
   EXPECT_EQ(args.GetString(GRPC_ARG_HTTP_CONNECT_SERVER), absl::nullopt);
 }
 
+// Test entries with CIDR blocks (Class A) in 'no_proxy' list.
+TEST(NoProxyTest, CIDRClassAEntries) {
+  ScopedSetEnv no_proxy("foo.com,192.168.0.255/8");
+  auto args = ChannelArgs().Set(GRPC_ARG_HTTP_PROXY, "http://proxy.google.com");
+  EXPECT_EQ(HttpProxyMapper().MapName("dns:///192.0.1.1:443", &args),
+            absl::nullopt);
+  EXPECT_EQ(args.GetString(GRPC_ARG_HTTP_CONNECT_SERVER), absl::nullopt);
+
```

Comment:
can you add some comments please - something like - "address not matching no_proxy cidr block"